libcloudforensics/providers/aws/forensics.py

```diff
@@ -445,3 +446,36 @@ def CopyEBSSnapshotToS3(
   else:
     logger.info(
       'Image copy timeout. The process may be ongoing, or might have failed.')
+
+def InstanceNetworkQuarantine(zone: str,
+    instance_id: str,
+    exempted_src_subnets: Optional[List[str]] = None
+    ) -> None:
+  """Put an AWS EC2 instance in network quarantine.
+
+  Network quarantine is imposed via applying empty security groups to the
+  instance.
+
+  Args:
+    instance_id (str): : The id (i-xxxxxx) of the virtual machine.
+    exempted_src_subnets (List[str]): List of subnets that will be permitted
+  """
+  # Add /32 to any specified subnets that don't have a mask
+  # We're not checking the subnet is well formed, CreateIsolationSecurityGroup
+  # will take care of that
+  if exempted_src_subnets:
+    exempted_src_subnets[:] = [subnet if subnet.__contains__('/') else
+      subnet + '/32' for subnet in exempted_src_subnets]
+
+  try:
+    aws_account = account.AWSAccount(zone)
+    vpc = aws_account.ec2.GetInstanceById(instance_id).vpc
+    sg_id = \
+      aws_account.ec2.CreateIsolationSecurityGroup(vpc, exempted_src_subnets)
+    aws_account.ec2.SetInstanceSecurityGroup(instance_id, sg_id)
+  except errors.ResourceNotFoundError:
+    logger.error('Cannot qurantine non-existent instance')
```

Comment:
For each of these errors, maybe we can log the stacktrace? Otherwise it's hard to understand what and why it went wrong

Reply:
Looking at the other methods in `forensics.py` it looks to actually be standard to not catch exceptions at this level, and propagate them higher up the stack. So, adding a reraise for the instance not existing, and removing the other catches.